### PR TITLE
docs: added PR template

### PR DESCRIPTION
**Pull request by szenius**
This PR adds a template for pull requests. After this is merged, [all new pull requests will automatically have the template.](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/about-issue-and-pull-request-templates#pull-request-templates)

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [ ] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3)
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

**Review by sebastianquek — APPROVED** [review]
LGTM!

**Review by yongjiajie — APPROVED** [review]
Looks great, thanks for doing this 💯

**Review by millasml — APPROVED** [review]
LGTM!

**Review by dalsontws — APPROVED** [review]
LGTM! 🚢  Thanks!!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+This PR adds... <!-- A brief description of what your PR does -->
+
+<!-- This checklist is just a guideline.-->
+<!-- If any of the following does not apply to your PR, please leave them unchecked and explain in the PR description -->
+
+- [ ] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
+- [ ] I've requested reviews from 1-2 reviewers
+- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
+- [ ] I've added/updated unit/integration tests
+- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3)
```
Comment (ilhaoni) [inline .github/pull_request_template.md:12]: hmm, do you think we should specify that it's the e2e tests for just the happy flow? :O would that be clearer haha

**Review by maksha19 — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (szenius) [reply .github/pull_request_template.md]: Added, wrote "at least the happy flow" since it's OK to also write for the unhappy flow

Comment (ilhaoni) [reply .github/pull_request_template.md]: nice, thanks!! :D

**Review by ilhaoni — APPROVED** [review]
LGTM! :D Thanks for adding this~~

**Pull request merged**